admin/includes/configuration_checks.php

```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * @package admin
+ * @copyright Copyright 2015 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version 
+ */
+
+  // Function used for configuration checks only
+  function check_configuration($variable, $check_string) { 
+     $parts = explode(",", $check_string, 2); 
+     eval('$id = ' . $parts[0] . ';'); 
+     eval('$options = ' . $parts[1] . ';'); 
```

Comment:
Is it at all possible to avoid using `eval()` here?

I know there are some web apps that use eval() generously, but it's also a potential sec urity hole target, especially when its inputs are coming directly from the db.  I know ZC uses it in a couple places, but we try hard to limit those where possible.

I wonder if the `$id` one can be read as 
```php
$id=constant($parts[0]);
```
And then check whether its value is a sane value before passing it along to `filter_var`.

Not sure about `$parts[1]` though. Running it through a sanitizing regex first might be doable, but there are so many characters that are potentially acceptable that it might be hard to come up with a blacklist of mal icious characters to exclude.